source/zarchive.d

```diff
@@ -236,6 +238,50 @@ public:
         return false;
     }
 
+    void extractArchive (const string dest)
+    in
+    {
+        import std.file : isDir;
+        assert (dest.isDir);
+    }
+    body
+    {
+        import std.file : chdir, getcwd;
+
+        archive *ar;
+        archive *ar_dest;
+        archive_entry *en;
+
+        const void *buf;
+        size_t size;
+        long offset;
+
+        string cwd;
+
+        cwd = getcwd ();
+
+        ar = openArchive ();
+        scope(exit) archive_read_free (ar);
+
+        dest.chdir;
+        scope(exit) cwd.chdir;
```

Comment:
That's why I never use the disk writer of libarchive - actually, the archive decompressor classes should have all you need already.
There is also on extract-by-regex function which you could probably use (would need a flag to preserve directory structures, I think)